src/System.Linq/tests/SelectManyTests.cs

```diff
@@ -452,10 +452,21 @@ public void DisposeAfterEnumeration(int sourceLength, int subLength)
             Assert.All(subState, s => Assert.Equal(subLength + 1, s));
             Assert.All(subCollectionDisposed, t => Assert.True(t));
 
-            // Make sure the iterator's enumerator has been disposed properly.
-            Assert.Equal(0, e.Current); // Default value.
-            Assert.False(e.MoveNext());
-            Assert.Equal(0, e.Current);
+            // .NET Core fixes an oversight where we wouldn't properly dispose
+            // the SelectMany iterator. See https://github.com/dotnet/corefx/pull/13942.
```

Comment:
I think this is just result of `Iterator` erasing `Current` in its dispose. Most hand-written enumerators in Linq would have this behavior. 
I do not think full FX had a bug here, compiler-generated should call Dispose as well, but it does not clear `Current`.

I think these sort of changes are ok. They happen when `foreach` iterator is manually implemented.

---
In reply to: [109670698](https://github.com/dotnet/corefx/pull/17877#discussion_r109670698) [](ancestors = 109670698)